### PR TITLE
Pip: Do not raise PathDependenciesNotReachable for missing setup.py

### PR DESCRIPTION
**Pull request by jurre**
Previously, Dependabot would assume that for every requirements.txt with
a self-reference (`-e .`), we should expect a `setup.py` to be present. The
self-reference is considered a path dependency that needs a setup.py.

However, nowadays it's valid to have a requirements.txt without a
setup.py. This change accounts for that by still attempting to pull in a
setup.py if it exists, but to not raise and error when it doesn't.

Closes https://github.com/dependabot/dependabot-core/pull/5324
Resolves https://github.com/dependabot/dependabot-core/issues/4483

### CODE REVIEW

File: python/spec/dependabot/python/file_fetcher_spec.rb [hunk 1]
```diff
@@ -1037,25 +1037,44 @@
...
+            headers: { "content-type" => "application/json" }
+          )
+        stub_request(:get, url + "pyproject.toml?ref=sha").
+          with(headers: { "Authorization" => "token token" }).
+          to_return(
+            status: 200,
+            body: fixture("github", "contents_python_pyproject.json"),
+            headers: { "content-type" => "application/json" }
+          )
+        stub_request(:get, url + "setup.cfg?ref=sha").
```
Comment (rouge8) [inline python/spec/dependabot/python/file_fetcher_spec.rb:1061]: There should probably be a test for no `setup.cfg` as well, just a `requirements.txt` with `-e .` and `pyproject.toml`

Comment (jurre) [reply python/spec/dependabot/python/file_fetcher_spec.rb:1061]: I don't think another test would add much extra coverage, I would not assume that to be able to break individually from this current test, or am I missing something?

**Review by DanielNoord — COMMENTED** [review]
As far as I can see this should work. Thanks for all the work you did here @jurre 😄

**Review by jakecoffman — APPROVED** [review]

File: python/spec/dependabot/python/file_fetcher_spec.rb (hunk 1 above)
Comment (jurre) [reply python/spec/dependabot/python/file_fetcher_spec.rb:1061]: Thanks for checking this out though, always appreciate another set of eyes! 🙇

**Review by pavera — APPROVED** [review]

**Comment by DanielNoord** [comment]
Thanks @jurre this is now working as expected 😄

**Pull request merged**